File: CoreSupply.Ordering.API/Domain/Entities/Order.cs
using CoreSupply.BuildingBlocks.DDD;

namespace CoreSupply.Ordering.API.Domain.Entities
{
    // این کلاس اصلی سفارش است
    public class Order : Entity<Guid>
    {
        public string UserName { get; set; }
        public decimal TotalPrice { get; set; }

        // آدرس (Billing Address)
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string EmailAddress { get; set; }
        public string AddressLine { get; set; }
        public string Country { get; set; }
        public string State { get; set; }
        public string ZipCode { get; set; }

        // پرداخت
        public string CardName { get; set; }
        public string CardNumber { get; set; }
        public string Expiration { get; set; }
        public string CVV { get; set; }
        public int PaymentMethod { get; set; }

        // وضعیت سفارش (یک Enum ساده)
        public string Status { get; set; } = "Pending"; // در آینده تبدیل به Enum می‌شود
    }
}

File: frontend/src/domain/models/Order.ts
// src/domain/models/Order.ts

// اینترفیس برای ثبت سفارش (بدون تغییر)
export interface BasketCheckout {
  userName: string;
  totalPrice: number;
  
  // اطلاعات آدرس
  firstName: string;
  lastName: string;
  emailAddress: string;
  addressLine: string;
  country: string;
  state?: string;
  zipCode: string;

  // اطلاعات پرداخت (فیک)
  cardName: string;
  cardNumber: string;
  expiration: string;
  cvv: string;
  paymentMethod: number;
}

// اینترفیس برای نمایش تاریخچه سفارش (بدون تغییر)
export interface OrderResponse {
  userName: string;
  totalPrice: number;
  firstName: string;
  lastName: string;
  emailAddress: string;
  addressLine: string;
  country: string;
  state: string;
  zipCode: string;
}

export interface OrderSummary {
  id: string;          // مطابق خروجی بک‌اِند
  createdAt: string;   // مطابق خروجی بک‌اِند
  status: string;      // مطابق خروجی بک‌اِند
  totalPrice: number;  // مطابق خروجی بک‌اِند
}
